documentation/naming-and-structure.asciidoc

```diff
@@ -12,18 +12,26 @@ Scope: Backend microservices providing HTTP/RESTful API.
 * introduce complexity progresively, if required
 * pragmatic and productive, recommendations not rules
 
+=== Packaging
 
-`«group».«artifact»[.«component»].«layer»[.«type»]*`
+The new packaging schema is suggested as:
+```
+«root».«component».«layer»[.«detail»]*
+```
+
+See also existing https://github.com/devonfw/devon4j/blob/master/documentation/coding-conventions.asciidoc#packages[devon4j packaging] that we want to simplify and modernize here.
 
 .Segments of package schema
 [options="header"]
 |=============================================
 | *Segment*      | *Description* | *Example*
-| «group» | Matches your maven `groupId` coordinate, basic name-space of the organization or IT project owning the code. Corresponds to domain name reversed. |`com.devonfw`
-| «artifact» | Matches your maven `artifact` coordinate, converted to java package conventions(e.g. `-` omitted). Can be further simplified if it does not cause disambiguity(e.g. `presales-order-bff`-> `pobff`) | `orderservice`, `usermgmt`..
-| «component» | Only used if the service serves multiple business components. Most microservices usually address a single business component/domain. In this case it is ommited(its already implied by the artifactid)| `users`, `order`
-| «layer» | The name of the technical layer (See link:architecture.asciidoc[technical architecture]) which is one of the predefined layers (`domain`, `rest`, `logic`) or `common` for code not assigned to a technical layer (datatypes, cross-cutting concerns). Additional layers can be introduced if they have clear scope e.g. `batch`, `process`.. | `rest`
-| «type» | Further division based on common stereotypes based on type of component. Suggested stereotypes for `rest` layer are [`controller`, `model`, `mapper`, `filter`..] for `domain` layer: [`model`, `dao`]  | `dao`
+| `«root»` | The root package. Typically we suggest to use `«group».«artifact»` for `«root»`. However, we will not enforce anything and let you the freedom of choice if the recommendation does not fit for you.|`com.devonfw`
+| `«group»` | Matches your maven `groupId` coordinate, basic name-space of the organization or IT project owning the code. Corresponds to domain name reversed. |`com.devonfw`
```

Comment:
Since group and artifact have been dropped above, do we still need the explanation of it at this point?

Reply:
we suggest to use `«group».«artifact»` as `«root»` so in many cases it will remain as it was before.
However, with the change, I want to underlying that this is just a recommendation and not a "regex" that is enforced.
We have to be very specific as we have tools (CobiGen, sonar-devon4j-plugin, etc.) that need to have to follow strict rules as they are just tools and no AI. IMHO with TKit you never had such issues to consider... You are also inside one single customer domain where it might suite to have maven `groupId` equal to the Java toplevel packages. However, I can tell you from the start that we will have projects where this is for historical reasons unacceptable and I do not want to lose any project whilst reading such packaging convention :)